tests/providers/amazon/aws/sensors/test_emr_serverless.py

```diff
@@ -0,0 +1,119 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from unittest.mock import MagicMock
+
+import pytest
+
+from airflow.exceptions import AirflowException, AirflowSkipException
+from airflow.providers.amazon.aws.sensors.emr import EmrServerlessJobSensor
+
+
+class TestEmrServerlessJobSensor:
+    def setup_method(self):
+        self.app_id = "vzwemreks"
+        self.job_run_id = "job1234"
+        self.sensor = EmrServerlessJobSensor(
+            task_id="test_emrcontainer_sensor",
+            application_id=self.app_id,
+            job_run_id=self.job_run_id,
+            aws_conn_id="aws_default",
+        )
+
+    def set_get_job_run_return_value(self, return_value: dict[str, str]):
+        self.mock_hook = MagicMock()
+        self.mock_hook.conn.get_job_run.return_value = return_value
+        self.sensor.hook = self.mock_hook
+
+    def assert_get_job_run_was_called_once_with_app_and_run_id(self):
+        self.mock_hook.conn.get_job_run.assert_called_once_with(
+            applicationId=self.app_id, jobRunId=self.job_run_id
+        )
+
+
+class TestPokeReturnsFalse(TestEmrServerlessJobSensor):
+    def run_test(self, get_job_run_return_value: dict[str, str]):
+        self.set_get_job_run_return_value(get_job_run_return_value)
+        assert not self.sensor.poke(None)
+        self.assert_get_job_run_was_called_once_with_app_and_run_id()
+
+    def test_when_state_is_pending_poke_should_return_false(self):
+        self.run_test(get_job_run_return_value={"jobRun": {"state": "PENDING"}})
+
+    def test_when_state_is_running_poke_should_return_false(self):
+        self.run_test(get_job_run_return_value={"jobRun": {"state": "RUNNING"}})
+
+    def test_when_state_is_scheduled_poke_should_return_false(self):
+        self.run_test(get_job_run_return_value={"jobRun": {"state": "SCHEDULED"}})
+
+    def test_when_state_is_submitted_poke_should_return_false(self):
+        self.run_test(get_job_run_return_value={"jobRun": {"state": "SUBMITTED"}})
+
+
+class TestPokeReturnsTrue(TestEmrServerlessJobSensor):
+    def test_when_state_is_success_poke_should_return_true(self):
+        self.set_get_job_run_return_value({"jobRun": {"state": "SUCCESS"}})
+        assert self.sensor.poke(None)
+        self.assert_get_job_run_was_called_once_with_app_and_run_id()
```

Comment:
You could include this test as part of the test above using [parametrize](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html)

Reply:
Same answer as above.